vernac/comArguments.ml

```diff
@@ -246,17 +247,21 @@ let vernac_arguments ~section_local reference args more_implicits flags =
   let red_behavior =
     let open Reductionops.ReductionBehaviour in
     match never_unfold_flag, nomatch_flag, rargs, nargs_for_red with
-    | true, false, [], None -> Some NeverUnfold
+    | true, false, [], None -> if clear_red_flag then err_incompat "simpl never" "clear simpl" else Some NeverUnfold
     | true, true, _, _ -> err_incompat "simpl never" "simpl nomatch"
     | true, _, _::_, _ -> err_incompat "simpl never" "!"
     | true, _, _, Some _ -> err_incompat "simpl never" "/"
     | false, false, [], None  -> None
-    | false, false, _, _ -> Some (UnfoldWhen { nargs = nargs_for_red;
-                                               recargs = rargs;
-                                             })
-    | false, true, _, _ -> Some (UnfoldWhenNoMatch { nargs = nargs_for_red;
-                                                     recargs = rargs;
-                                                   })
+    | false, false, _, _ ->
+      if clear_red_flag then err_incompat "clear simpl" "/" else
```

Comment:
Shouldn't we distinguish `!` and `/` here? (as two or three lines above)

Reply:
Right. Updated.

This actually raises the question of whether the same effect as `clear simpl` can be obtained by issueing `Arguments` with a `!` on exactly the structurally decreasing argument of a global fix/cofix and by adding `!` on the argument that would expose a `match` otherwise. I could not find a counterexample for `simpl` (yet) but I could find one for `cbn`:
```coq
Eval cbn in pred (id 0).  (* 0 *)
Arguments id A x : simpl never. 
Eval cbn in pred (id 0).  (* pred (id 0) *)
Arguments id A !x.
Eval cbn in pred (id 0). (* id 0 *)
```